packages/client/src/sw.ts

```diff
@@ -0,0 +1,131 @@
+const CACHE_NAME = 'echo-runner-cache-v1';
+const API_CACHE_NAME = 'echo-runner-api-cache-v2';
+const API_ENDPOINTS = ['/auth/user'];
+const AUTH_API_BASE = 'https://ya-praktikum.tech/api/v2/auth';
+
+const URLS = ['/index.html', '/', '/topics', '/signin', '/signup', '/profile', '/leaderboard', '/game', '/error'];
+
+self.addEventListener('install', ((event: ExtendableEvent) => {
+  console.log('🔧 Service Worker installing...');
+  event.waitUntil(
+    Promise.all([
+      caches.open(CACHE_NAME).then((cache) => {
+        console.log('Opened static cache:', CACHE_NAME);
+        return cache.addAll(URLS);
+      }),
+      caches.open(API_CACHE_NAME).then((cache) => {
+        console.log('Opened API cache:', API_CACHE_NAME);
+        return cache;
+      }),
+    ]).catch((err) => {
+      console.log('Install error:', err);
+      throw err;
+    }),
+  );
+}) as EventListener);
+
+self.addEventListener('activate', ((event: ExtendableEvent) => {
+  console.log('activate');
+  event.waitUntil(
+    caches.keys().then((cacheNames) => {
+      console.log('activate:', cacheNames);
+      return Promise.all(
+        cacheNames
+          .filter((name) => name !== CACHE_NAME && name !== API_CACHE_NAME)
+          .map((name) => {
+            caches.delete(name);
+            console.log('delete cache', name);
+          }),
+      );
+    }),
+  );
+}) as EventListener);
+
+self.addEventListener('fetch', ((event: FetchEvent) => {
+  if (isApiRequest(event.request.url)) {
+    event.respondWith(handleApiRequest(event));
+    return;
+  }
+
+  event.respondWith(
+    caches.match(event.request).then((response) => {
+      if (response) {
+        console.log('response found in cache');
+        return response;
+      }
+
+      const fetchRequest = event.request.clone();
+      return fetch(fetchRequest).then((response) => {
+        if (!response || response.status !== 200 || response.type !== 'basic') {
+          console.log('failed response');
+          return response;
+        }
+
+        const responseToCache = response.clone();
+        caches.open(CACHE_NAME).then((cache) => {
```

Comment:
Ты не обрабатываешь ошибки


```suggestion
caches.open(CACHE_NAME)
  .then((cache) => cache.put(event.request, responseToCache))
  .catch((err) => console.warn('Cache put failed:', err));
```

cache.put может упасть (например, если request не кешируемый), но .then() не содержит .catch.